### PR TITLE
Code change to retry based on retry-after header

### PR DESCRIPTION
**Pull request by sarangan12**
@veronicagg @devigned Please review and approve.

@salameer FYI...

### CODE REVIEW

File: runtime/ms_rest/lib/ms_rest/http_operation_request.rb [hunk 1]
```diff
@@ -77,14 +77,22 @@ def run_promise(&block)
...
-          yield(req) if block_given?
+        loop do
+          @response = @connection.run_request(:"#{method}", build_path, body, {'User-Agent' => user_agent}.merge(headers)) do |req|
+            req.params = req.params.merge(query_params.reject{|_, v| v.nil?}) unless query_params.nil?
+            yield(req) if block_given?
+          end
+
+          break if ((@response.status != 429) || (@response.status == 429 && @response.headers['retry-after'].nil?))
+
+          if(@response.status == 429 && !@response.headers['retry-after'].nil?)
```
Comment (veronicagg) [inline runtime/ms_rest/lib/ms_rest/http_operation_request.rb]: is this retry behavior general or Azure only? should it belong to ms_rest or ms_rest_azure?

**Comment by veronicagg** [comment]
@sarangan12 Ruby build drop is failing, please check that one before merging.

File: runtime/ms_rest/lib/ms_rest/http_operation_request.rb (hunk 1 above)
Comment (sarangan12) [reply runtime/ms_rest/lib/ms_rest/http_operation_request.rb]: The throttling seems to be general in all other SDKs. So, following the same.

**Comment by sarangan12** [comment]
Ruby Build drop is failing for a known issue. I will fix it soon.

**Pull request merged**